ant-networking/src/record_store_api.rs

```diff
@@ -119,51 +133,50 @@ impl UnifiedRecordStore {
         &self,
         key: &RecordKey,
         network_size: Option<u64>,
-    ) -> (QuotingMetrics, bool) {
+    ) -> Result<(QuotingMetrics, bool)> {
         match self {
             Self::Client(_) => {
-                warn!("Calling quoting metrics calculation at Client. This should not happen");
-                Default::default()
+                error!("Calling quoting_metrics at Client. This should not happen");
+                Err(NetworkError::OperationNotAllowedOnClientRecordStore)
             }
-            Self::Node(store) => store.quoting_metrics(key, network_size),
+            Self::Node(store) => Ok(store.quoting_metrics(key, network_size)),
         }
     }
 
     pub(crate) fn payment_received(&mut self) {
         match self {
             Self::Client(_) => {
-                warn!("Calling payment_received at Client. This should not happen");
+                error!("Calling payment_received at Client. This should not happen");
             }
             Self::Node(store) => store.payment_received(),
         }
     }
 
-    pub(crate) fn get_farthest_replication_distance(&self) -> Option<U256> {
+    pub(crate) fn get_farthest_replication_distance(&self) -> Result<Option<U256>> {
         match self {
-            Self::Client(_store) => {
-                warn!("Calling get_distance_range at Client. This should not happen");
-                None
+            Self::Client(_) => {
+                error!(
+                    "Calling get_farthest_replication_distance at Client. This should not happen"
+                );
+                Err(NetworkError::OperationNotAllowedOnClientRecordStore)
             }
-            Self::Node(store) => store.get_responsible_distance_range(),
+            Self::Node(store) => Ok(store.get_responsible_distance_range()),
         }
     }
 
     pub(crate) fn set_distance_range(&mut self, distance: U256) {
         match self {
-            Self::Client(_store) => {
-                warn!("Calling set_distance_range at Client. This should not happen");
+            Self::Client(_) => {
+                error!("Calling set_distance_range at Client. This should not happen");
             }
             Self::Node(store) => store.set_responsible_distance_range(distance),
         }
     }
 
-    pub(crate) fn get_farthest(&self) -> Option<RecordKey> {
+    pub(crate) fn get_farthest(&self) -> Result<Option<RecordKey>> {
```

Comment:
In what case do we have a `Ok(None)`?
Any chance we can simplify the return type to `Result<RecordKey>` or does the `Option` have any special meaning here?

Reply:
It is None when the record store is empty / when the farthest record is not calculated. It is from this field.
```
    /// Farthest record to self
    farthest_record: Option<(Key, Distance)>,
```